### PR TITLE
[main] Revert changes to hosted providers gke and eks clusterconfigs

### PR DESCRIPTION
**Pull request by cpinjani**
Backport: https://github.com/rancher/shepherd/pull/290

## Problem
To be able to have empty arrays and maps in the rest calls.
Ref: https://github.com/rancher/rancher/pull/39034

## Solution
- This PR is to revert the changes made to hosted clusterconfigs in https://github.com/rancher/shepherd/pull/219
- (A small change to make _cmekConfig_ field optional is also covered)

### CODE REVIEW

**Comment by igomez06** [comment]
@cpinjani A schema change is not necessary? Did you manually update the generated files or use the code generator?

**Comment by cpinjani** [comment]
@igomez06 Schema changes were inessential, I have manually updated the files for revert.

**Comment by igomez06** [comment]
> @igomez06 Schema changes were inessential, I have manually updated the files for revert.

The worry is if we run the generator code it change it again, we can't manually change these files. The schema changes are then necessary. Please take a look at this file. https://github.com/rancher/shepherd/blob/main/pkg/schemas/management.cattle.io/v3/schema.go

Try this locally what happens when we run `go generate ./` your changes are reverted again?

**Comment by cpinjani** [comment]
Yes my concern as well, running `go generate ./` reverts this PR changes. And hosted configs require custom schema for some fields as implemented in https://github.com/rancher/rancher/pull/39034

**Comment by igomez06** [comment]
> Yes my concern as well, running `go generate ./` reverts this PR changes. And hosted configs require custom schema for some fields as implemented in [rancher/rancher#39034](https://github.com/rancher/rancher/pull/39034)

Then please take the time to review the schema code and update as needed. If you need further assistance we can set up a time and look at it together. Thank you!

**Comment by cpinjani** [comment]
@igomez06 Thanks for pointing to the removed schema code. Its reverted and PR is ready for review.

File: extensions/clusters/gke/gke_cluster_config.go [hunk 1]
```diff
@@ -143,10 +143,14 @@ func clusterAddonsBuilder(clusterAddons *ClusterAddons) *management.GKEClusterAd
 }
 
 func cmekConfigBuilder(cmekConfig *CMEKConfig) *management.CMEKConfig {
-	return &management.CMEKConfig{
-		KeyName:  cmekConfig.KeyName,
-		RingName: cmekConfig.RingName,
+	var CMEKConfig *management.CMEKConfig
+	if cmekConfig != nil {
```
Comment (caliskanugur) [inline extensions/clusters/gke/gke_cluster_config.go:147]: Is there a chance that this is a nil? Is this something new?

Comment (cpinjani) [reply extensions/clusters/gke/gke_cluster_config.go:147]: Yes, this change allows to omit specifying _customerManagedEncryptionKey_ in CATTLE_TEST_CONFIG's gkeClusterConfig  section.

File: extensions/clusters/gke/gke_cluster_config.go [hunk 2]
```diff
@@ -143,10 +143,14 @@ func clusterAddonsBuilder(clusterAddons *ClusterAddons) *management.GKEClusterAd
...
+		CMEKConfig = &management.CMEKConfig{
+			KeyName:  cmekConfig.KeyName,
+			RingName: cmekConfig.RingName,
+		}
 	}
+	return CMEKConfig
```
Comment (caliskanugur) [inline extensions/clusters/gke/gke_cluster_config.go:153]: if we're just adding a nil check, do we have to add a new variable, could we do this way with less addition?

```suggestion

	if cmekConfig == nil {
		return nil
	}
	
	return  &management.CMEKConfig{
 			KeyName:  cmekConfig.KeyName,
 			RingName: cmekConfig.RingName,
 	}
	
```

Comment (igomez06) [reply extensions/clusters/gke/gke_cluster_config.go:153]: This is much cleaner way to do it.

Comment (cpinjani) [reply extensions/clusters/gke/gke_cluster_config.go:153]: Yes, this works. Thanks Ugur!

**Review by igomez06 — APPROVED** [review]
LGTM

**Comment by caliskanugur** [comment]
Chandan one last thing before this one gets merged, could you please check the PR history? I think accidentally one of my commits is included while rebasing


2.9 PR seems good, only this one

**Review by caliskanugur — APPROVED** [review]
LGTM

**Pull request merged**